assets/aml-benchmark/components/src/aml_benchmark/prompt_crafter/package/prompt_factory.py

```diff
@@ -303,15 +303,32 @@ def create_prompt(self, row: Dict) -> ChatPrompt:
                                                Role.system.name)
             messages.extend(system_message)
 
+        prefix_messages = []
         if self.prefix:
             prefix_messages = self._parse_affix(affix=self.prefix)
-            messages.extend(prefix_messages)
+
+        # Parsing the prompt pattern
+        input_messages = self._parse_affix(affix=self.prompt_pattern)
 
         if self.few_shot_pool is not None and self.n_shots > 0:
             few_shot_messages = self._create_few_shots(row)
+            if len(prefix_messages) == 1 and len(few_shot_messages) > 1:
```

Comment:
Why are we checking if len few_shot_messages > 1? For n_shots = 1, will we handle it differently?

Reply:
No. Updated it. Thanks!